service/src/main/java/bio/terra/catalog/service/ExternalSystemService.java

```diff
@@ -32,6 +32,10 @@ public Map<String, StorageSystemInformation> getDatasets() {
                 dataset -> new StorageSystemInformation(DatasetAccessLevel.DISCOVERER, null)));
   }
 
+  public StorageSystemInformation getDataset(String storageSourceId) {
+    return new StorageSystemInformation(getRole(storageSourceId), null);
```

Comment:
This form of calling `StorageSystemInformation(x, null)` is used in two places now. Can you add a constructor override that only accepts a single parameter for this usage? Looking at all usages of this object's constructor I see two cases:
- both access level and phs id
- only access level

The current definition uses a no-arg constructor to support a builder-like pattern. But since it's only ever used in these two ways, I think it would be better to remove the builder pattern for now, and add a custom constructor for the `only access level` case. The other case is already handled by the `record` provided constructor.